### ISSUE TITLE
"no suitable plugins found" message is not quite helpful

### ISSUE THREAD

**Issue by genodeftest** [exaile/exaile#35]
I was unable to play MP3 files and got this message in GUI: `no suitable plugins found`. I didn't know why. I don't think a normal user would find out what to do in this case.

On console I got this message:

```
<gst.Message GstMessageError, gerror=(GError)NULL, debug=(string)"gstdecodebin2.c\(3576\):\ gst_decode_bin_expose\ \(\):\ /GstPlayBin2:player/GstURIDecodeBin:uridecodebin3/GstDecodeBin2:decodebin23:\012no\ suitable\ plugins\ found"; from decodebin23 at 0x7ff0b4068b10> ['__class__', '__cmp__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__grefcount__', '__gstminiobject_init__', '__gtype__', '__hash__', '__init__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'copy', 'flags', 'parse_async_start', 'parse_buffering', 'parse_buffering_stats', 'parse_clock_lost', 'parse_clock_provide', 'parse_duration', 'parse_error', 'parse_info', 'parse_new_clock', 'parse_qos', 'parse_qos_stats', 'parse_qos_values', 'parse_request_state', 'parse_segment_done', 'parse_segment_start', 'parse_state_changed', 'parse_step_done', 'parse_step_start', 'parse_stream_status', 'parse_structure_change', 'parse_tag', 'parse_tag_full', 'parse_warning', 'set_buffering_stats', 'set_qos_stats', 'set_qos_values', 'set_seqnum', 'src', 'structure', 'timestamp', 'type'] (xl.player._base)
```

After installing gstreamer-plugins-ugly this error message was gone.

It would be useful to add a hint that the user should install gstreamer plugins and possibly add the file format. And/or a link to https://exaile.readthedocs.org/en/latest/user/faq.html#error-no-suitable-plugin-found-when-playing-a-mp3-m4a-etc

**Comment by genodeftest** [comment]
for m4a files this is gstreamer-ffmpeg
for wma files this is gstreamer-plugins-bad
for aac there seems to be no plugin for gstreamer 0.10

For more details see 
http://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-good-plugins/html/
http://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-bad-plugins/html/
http://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-ugly-plugins/html/
and
http://gstreamer.freedesktop.org/documentation/plugins.html

**Comment by virtuald** [comment]
Did you mean to close this issue?

**Comment by genodeftest** [comment]
No, I didn't. Thanks.

**Comment by genodeftest** [comment]
The error message changed to `Playback error encountered: Your GStreamer installation is missing a plug-in`
It would be nice to get more information. This is possible, see the [Base Utils Library](http://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-base-libs/html/gstreamer-base-utils.html) provided by the gstreamer-plugins-base package. It is possible to prompt the user to install missing plugins, e.g. by using PackageKit or gnome-software. This does not require extra packages to be installed. [Here](https://git.gnome.org/browse/totem/tree/src/backend/bacon-video-widget-gst-missing-plugins.c) is how totem (gnome video application) handles it.

**Comment by virtuald** [comment]
Quod Libet has some code to do it also.

**Comment by Francewhoa** [comment]
Same error message here
![error_no suitable plugins found](https://cloud.githubusercontent.com/assets/119942/25100039/84ce610e-2363-11e7-838f-d8cb3159b031.png)

It was hard to find out how to fix that. The error message is not clear for beginner users.

I'm using Exaile 3.4.0. Debian 8 Jessie. GNOME 3.14.1. All of the above fresh installed.

genodeftest's suggestion in [that comment #35](https://github.com/exaile/exaile/issues/35#issuecomment-152653308) above sounds good. 

Another option to consider is when packaging Exaile for Debian, somehow during that package installation the user could be prompt about optionally installing all those packages dependencies? VLC does something like that, and it works for most codecs cases. I'm referring to the packages dependencies or any other dependencies that could be batch install with that command: `apt-get install gstreamer1.0-plugins-*` 
Those packages are [listed here](https://packages.debian.org/search?searchon=names&keywords=gstreamer1.0-plugins) for all Debian version, architecture, and suite. 

Any other suggestion to resolve that challenge?

Me and my team would be happy to contribute testing and documentation for such improvement

**Comment by virtuald** [comment]
@Francewhoa at the moment I think my priority is to just get Exaile 4.0 out the door... it's been way too long (2 years now?), and there's only a few critical issues left before I'm comfortable calling it a release. :) I'm hoping to get those taken care of in the next month or two -- hopefully life won't get in the way again! :)

I agree that utilizing the GStreamer utilities to give the user a better error message would be great, and shouldn't be particularly difficult to do.

**Comment by Francewhoa** [comment]
Thanks @virtuald :) Sounds great. I look forward to trying Exaile 4.0. I would be happy to contribute testing and documentation then if needed. 

I love Exaile. One of my favourite music player. It's easy to use. Cross-platform. Lots of features. Lightweight. Also more secure because its code is publicly available for review and contributions.

Thanks for all your contributions

**Comment by genodeftest** [comment]
I'm working on a solution based on GstPbutils, a GObject Introspection enabled library provided by GStreamer plugins base. Can we rely on this being present?

Can anyone confirm that it can be loaded on Windows or Mac OS? Please run these lines of code in a python console from Exaile:
```
import gi

from gi.repository import Gst
gi.require_version('GstPbutils', '1.0')
from gi.repository import GstPbutils
GstPbutils.pb_utils_init()
```
that should be enough to test.

And is there a way to install GStreamer plugins on Windows or Mac OS?

**Comment by dangmai** [comment]
I ran that code fine in MSYS2 on Windows. Not sure about OS X since I don't have one.

**Comment by virtuald** [comment]
I would recommend looking at how quodlibet deals with this: https://github.com/quodlibet/quodlibet/blob/master/quodlibet/quodlibet/player/gstbe/player.py#L548

Because we're freezing the environment and controlling its setup, we can definitely guarantee that the library will be present in OSX and Windows. I tried it and it worked on OSX.

**Comment by genodeftest** [comment]
> I ran that code fine in MSYS2 on Windows. Not sure about OS X since I don't have one.

> I tried it and it worked on OS X.

Thanks. 

> […] quodlibet […]

They are also using Gst.Pbutils.

**Comment by genodeftest** [comment]
Fixed in #336.
